lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -425,7 +432,10 @@ def exec_module(self, **kwargs):
         # parse the virtual network resource group and name
         virtual_network_dict = parse_resource_id(self.virtual_network_name)
         virtual_network_name = virtual_network_dict.get('name')
-        virtual_network_resource_group = virtual_network_dict.get('resource_group', self.resource_group)
```

Comment:
The origin logic is 
```
dict = parse_virtual_network_id()
virtual_network_resource_group = dict[resource_group] || self.resource_group
```
And now it is changed to 
```
dict = parse_virtual_network_id()
virtual_network_resource_group = dict[resource_group] || self.virtual_network_resource_group || self.resource_group
```

I wonder why this extra logic is needed? Parsing id shares the same behavior like Azure-CLI

Reply:
dict[resource_group] will work if user specifies entire path/id of virtual_network, but if they want to specify virtual_network from another resource group, specifying virtual_network_resource_group will be simplified way to do this.
This pattern is also implemented in azure_rm_virtualmachine.

Reply:
It is okay, here I would like to make the precedence more clearly.

Reply:
I agree, the documentation should be clearer around the precedence, i.e. `virtual_network_resource_group` overrides `resource_group` and so on.